### PR TITLE
feat: remove powered by quintype from the footer

### PR DESCRIPTION
**Pull request by cricket-lover**
Reference: https://github.com/quintype/ace-planning/issues/710

### CODE REVIEW

File: src/templates/generic-story/generic-story.tsx [hunk 1]
```diff
@@ -44,18 +44,22 @@ const Wrapper = styled.div`
...
   const footerText = get(config, ["publisherConfig", "publisher-settings", "copyright"], null);
+  let showPoweredByQt = get(config, ["opts", "featureConfig", "showPoweredByQt"], true);
   const infiniteScrollInlineConfig = get(
     config,
     ["opts", "featureConfig", "infiniteScroll", "infiniteScrollInlineConfig"],
     null
   );
-  let lastComponent = <Footer text={footerText} />;
+
+  showPoweredByQt = typeof showPoweredByQt === "function" ? showPoweredByQt() : showPoweredByQt;
```
Comment (Jeevan-Kishore) [inline src/templates/generic-story/generic-story.tsx:54]: if `showPoweredByQt` is a function, wont it need parameters such as config to consume?

**Review by Jeevan-Kishore — CHANGES_REQUESTED** [review]

File: src/templates/generic-story/generic-story.tsx [hunk 2]
```diff
@@ -44,18 +44,22 @@ const Wrapper = styled.div`
  */
 export const GenericStory = ({ story, config }: CommonTemplateTypes) => {
   const footerText = get(config, ["publisherConfig", "publisher-settings", "copyright"], null);
+  let showPoweredByQt = get(config, ["opts", "featureConfig", "showPoweredByQt"], true);
```
Comment (ags1773) [inline src/templates/generic-story/generic-story.tsx:47]: Don't pass this as prop to footer. Check inside footer component

Comment (cricket-lover) [reply src/templates/generic-story/generic-story.tsx:47]: then we will have to pass config to footer.

Comment (ags1773) [reply src/templates/generic-story/generic-story.tsx:47]: context. Check how other components use context

File: src/atoms/footer/footer.tsx [hunk 3]
```diff
@@ -22,23 +24,26 @@ const PoweredBy = styled.a`
...
 `;
 
-const StyledFooter = styled("footer")<FooterTypes>`
+const StyledFooter = styled("footer") <FooterTypes>`
   ${(props) => genStyles(baseStyles, props.style, props)}
 `;
 
-const BaseFooter = (props: FooterTypes & { theme?: DefaultTheme }) => {
-  const { text, children, style } = props;
+const BaseFooter = ({ text, children, style, config, theme }: FooterTypes) => {
```
Comment (ags1773) [inline src/atoms/footer/footer.tsx:31]: add types. If `style` isn't being used, remove it

File: src/atoms/footer/footer.tsx [hunk 4]
```diff
@@ -22,23 +24,26 @@ const PoweredBy = styled.a`
...
+  let showPoweredByQt = get(config, ["opts", "featureConfig", "showPoweredByQt"], true);
 
+  showPoweredByQt = typeof showPoweredByQt === "function" ? showPoweredByQt(config) : showPoweredByQt;
```
Comment (ags1773) [inline src/atoms/footer/footer.tsx:34]: add types

File: src/atoms/footer/__snapshots__/footer.test.tsx.snap [hunk 5]
```diff
@@ -1,13 +1,3 @@
 // Jest Snapshot v1, https://goo.gl/fbAQLP
 
-exports[`Footer should render 1`] = `
```
Comment (ags1773) [inline src/atoms/footer/__snapshots__/footer.test.tsx.snap:3]: Add tests for:
- when nothing is passed (default)
- when footer is disabled and `showPoweredByQt` is a boolean
- when footer is disabled and `showPoweredByQt` is a function
- negative test cases

File: src/atoms/footer/footer.tsx (hunk 3 above)
Comment (cricket-lover) [reply src/atoms/footer/footer.tsx:31]: style is being used, do you mean theme?

**Review by ags1773 — APPROVED** [review]

**Pull request merged**